pkg/backup/main.go

```diff
@@ -69,70 +69,72 @@ func RequiredMissingError(name string) error {
-		return "", nil, fmt.Errorf("missing input parameter")
+	if params.VolumeName == "" || params.SnapshotName == "" || params.DestURL == "" {
+		return nil, nil, fmt.Errorf("missing input parameter")
 	}
 
-	if !util.ValidVolumeName(backupParams.VolumeName) {
-		return "", nil, fmt.Errorf("invalid volume name %v for backup %v", backupParams.VolumeName, backupParams.BackupName)
+	if !util.ValidVolumeName(params.VolumeName) {
+		return nil, nil, fmt.Errorf("invalid volume name %v for backup %v", params.VolumeName, params.BackupName)
 	}
 
 	var labelMap map[string]string
-	if backupParams.Labels != nil {
-		labelMap, err = util.ParseLabels(backupParams.Labels)
+	if params.Labels != nil {
+		labelMap, err = util.ParseLabels(params.Labels)
 		if err != nil {
-			return "", nil, errors.Wrapf(err, "cannot parse backup labels for backup %v", backupParams.BackupName)
+			return nil, nil, errors.Wrapf(err, "cannot parse backup labels for backup %v", params.BackupName)
 		}
 	}
 
 	volumeInfo, err := getVolumeInfoFromVolumeMeta()
 	if err != nil {
-		return "", nil, err
+		return nil, nil, err
 	}
 
 	backingFile, err := openBackingFile(volumeInfo.BackingFilePath)
 	if err != nil {
-		return "", nil, err
+		return nil, nil, err
 	}
 
-	replicaBackup := replica.NewBackup(backingFile)
+	backup := replica.NewBackup(params.BackupName, params.VolumeName, params.SnapshotName, backingFile)
 
 	volume := &backupstore.Volume{
-		Name:                 backupParams.VolumeName,
+		Name:                 params.VolumeName,
 		Size:                 volumeInfo.Size,
 		Labels:               labelMap,
-		BackingImageName:     backupParams.BackingImageName,
-		BackingImageChecksum: backupParams.BackingImageChecksum,
-		CompressionMethod:    backupParams.CompressionMethod,
+		BackingImageName:     params.BackingImageName,
+		BackingImageChecksum: params.BackingImageChecksum,
+		CompressionMethod:    params.CompressionMethod,
 		CreatedTime:          util.Now(),
 	}
 
 	snapshot := &backupstore.Snapshot{
-		Name:        backupParams.SnapshotName,
+		Name:        params.SnapshotName,
 		CreatedTime: util.Now(),
 	}
 
-	log.Infof("Starting backup for %v, snapshot %v, dest %v", volume, snapshot, backupParams.DestURL)
-
-	backupID, isIncremental, err := backupstore.CreateDeltaBlockBackup(&backupstore.DeltaBackupConfig{
-		BackupName:      backupParams.BackupName,
-		ConcurrentLimit: backupParams.ConcurrentLimit,
+	config := &backupstore.DeltaBackupConfig{
+		BackupName:      params.BackupName,
+		ConcurrentLimit: params.ConcurrentLimit,
 		Volume:          volume,
 		Snapshot:        snapshot,
-		DestURL:         backupParams.DestURL,
-		DeltaOps:        replicaBackup,
+		DestURL:         params.DestURL,
+		DeltaOps:        backup,
 		Labels:          labelMap,
-	})
+	}
+	return backup, config, nil
+}
+
+func DoBackupCreate(replicaBackup *replica.BackupStatus, config *backupstore.DeltaBackupConfig) error {
```

Comment:
nit: Ideally, for each exported function/method in any level (component or lib), I would recommend having logging at the beginning. For example, `Staring  backup creation...`.

note: Since we use go, so it's not possible to add this kind of logs automatically by using meta programming or AOP at runtime or compile time.

Reply:
Added. Thank you.